<launch>
  <arg name="sub_info_topic"  default="/depth_registered/camera_info"/>
  <arg name="sub_image_topic" default="/depth_registered/image_rect"/>

  <arg name="pub_info_topic"  default="/depth_interpolated/camera_info"/>
  <arg name="pub_image_topic" default="/depth_interpolated/image_raw"/>

  <node pkg="nodelet" type="nodelet" name="nodelet_manager" args="manager" />

  <node pkg="nodelet" type="nodelet" name="realsense_register" args="load depth_image_proc/register nodelet_manager -no-bond">
    <!-- Input -->
    <remap from="rgb/camera_info"   to="/camera/color/camera_info"/>
    <remap from="depth/camera_info" to="/camera/depth/camera_info"/>
    <remap from="depth/image_rect"  to="/camera/depth/image_rect_raw"/>
    <!-- Output -->
    <remap from="depth_registered/image_rect" to="/depth_registered/image_rect"/>
    <!-- Params -->
    <param name="queue_size" value="20"/>
    <!-- <param name="fill_upsampling_holes" value="true"/> -->
  </node>

  <node name="depth_interpolation" pkg="depth_interpolation" type="depth_interpolation.py" output="screen">
    <param name="sub_info_topic"  value="$(arg sub_info_topic)" type="string"/>
    <param name="sub_image_topic" value="$(arg sub_image_topic)" type="string"/>
    <param name="pub_info_topic"  value="$(arg pub_info_topic)" type="string"/>
    <param name="pub_image_topic" value="$(arg pub_image_topic)" type="string"/>
  </node>

</launch>